### DISCUSSION TITLE
How i parse this json value?

### DISCUSSION THREAD

**Discussion by SebnemSeymen** [serenity-is/Serenity#6750] [Help] [answered]
I have a json data that I have stored in my database. For example, I keep data such as why those who scored 5 got 5. I want to convert this json data into a string data type that the user can read and understand. I need to pull Description and GroupDescription data from this json and send it to my list. How can I do that?

[
  {
    "Group": "Group1",
    "RuleName": "Rule_1_1",
    "Description": "DİĞER",
    "GroupDescription": "KKB / Risk Merkezi kaydı yok / Onay vermedi",
    "Value": 1,
    "RuleParameters": "null"
  },
  {
    "Group": "Group2",
    "RuleName": "Rule_2_7",
    "Description": "DİĞER",
    "GroupDescription": "Bireysel Kredi/Konut Kredisi/Taşıt Kredisi/KK/KMH kullanımı",
    "Value": 1,
    "RuleParameters": "[{\"BireyselLimit\":{\"Range_1\":\"1500\",\"Range_2\":\"1500:3000\"}},{\"KrediTaksitTutari\":{\"Range_1\":\"500\",\"Range_2\":\"1000\"}}]"
  },
  {
    "Group": "Group3",
    "RuleName": "Rule_3_3",
    "Description": "Bireysel Takip Yok  / Ticari Takibi yok",
    "GroupDescription": "Bireysel Kredi/Konut Kredisi/Taşıt Kredisi/KK/KMH takip bilgisi(güncel)",
    "Value": 1,
    "RuleParameters": "[{\"Takip\":{\"Range_1\":\"5000\",\"Range_2\":\"500:5000\",\"Range_3\":\"0:500\"}}]"
  }, 
]

**Comment by minhhungit** [comment, marked as answer]
you can work with json easy by checking this https://learn.microsoft.com/en-us/sql/relational-databases/json/json-data-sql-server?view=sql-server-ver16

[Expression] attribute will work

**Reply by SebnemSeymen** [reply]
Thank you minhhungit , i try this query 

```
SELECT J.Description,J.GroupDescription,J.Value,NEWID() AS Id
FROM DealerScore C 
CROSS APPLY  OPENJSON(C.Result)
    WITH (		
			GroupDescription NVARCHAR(500) N'$."GroupDescription"',
            Description NVARCHAR(500) N'$."Description"',
			Value NVARCHAR(50) N'$."Value"'
		
         ) AS J 
WHERE ISJSON(C.Result) > 0 AND MUSTERIID='Customer Id Etc.'

```
this is not giving a error on db, when i try to execute my project, it is given error and not showing result.

showing console error;
Each data element must implement a unique 'MUSTERIID' property. Object at index '0' has repeated identity value 'Customer Id Etc.'


my sql code in Serenity Row ;

`[OuterApply("jDealerScore", "SELECT J.Description,J.GroupDescription,J.Value,NEWID() AS Id FROM DealerScore C CROSS APPLY  OPENJSON(C.Result)    WITH (GroupDescription NVARCHAR(500) N'$.\"GroupDescription\"',  Description NVARCHAR(500) N'$.\"Description\"',Value NVARCHAR(50) N'$.\"Value\"'    ) AS J WHERE ISJSON(C.Result) > 0 AND C.MUSTERIID=T0.MUSTERIID AND Scope='DRD'")]`

I'll be happy if you can help me.

**Reply by VictorTomaili** [reply]
Please be carefully while posting, ensure to use code blocks to share code, and ensure there is no private/secure data exist in your post.

**Reply by SebnemSeymen** [reply]
Thank you for your warning, I have made the necessary corrections. Do you have an idea about the solution?
